sklearn/inspection/partial_dependence.py

```diff
@@ -598,71 +620,12 @@ def convert_feature(fx):
-
-    if fig is None:
-        fig = plt.figure()
-    else:
-        fig.clear()
-
-    if line_kw is None:
-        line_kw = {'color': 'green'}
-    if contour_kw is None:
-        contour_kw = {}
-
-    n_cols = min(n_cols, len(features))
-    n_rows = int(np.ceil(len(features) / float(n_cols)))
-    axs = []
-    for i, fx, name, (avg_preds, values) in zip(
-            count(), features, names, pd_result):
-        ax = fig.add_subplot(n_rows, n_cols, i + 1)
-
-        if len(values) == 1:
-            ax.plot(values[0], avg_preds[target_idx].ravel(), **line_kw)
-        else:
-            # make contour plot
-            assert len(values) == 2
-            XX, YY = np.meshgrid(values[0], values[1])
-            Z = avg_preds[target_idx].T
-            CS = ax.contour(XX, YY, Z, levels=Z_level, linewidths=0.5,
-                            colors='k')
-            ax.contourf(XX, YY, Z, levels=Z_level, vmax=Z_level[-1],
-                        vmin=Z_level[0], alpha=0.75, **contour_kw)
-            ax.clabel(CS, fmt='%2.2f', colors='k', fontsize=10, inline=True)
-
-        # plot data deciles + axes labels
-        deciles = mquantiles(X[:, fx[0]], prob=np.arange(0.1, 1.0, 0.1))
-        trans = transforms.blended_transform_factory(ax.transData,
-                                                     ax.transAxes)
-        ylim = ax.get_ylim()
-        ax.vlines(deciles, [0], 0.05, transform=trans, color='k')
-        ax.set_xlabel(name[0])
-        ax.set_ylim(ylim)
-
-        # prevent x-axis ticks from overlapping
-        ax.xaxis.set_major_locator(MaxNLocator(nbins=6, prune='lower'))
-        tick_formatter = ScalarFormatter()
-        tick_formatter.set_powerlimits((-3, 4))
-        ax.xaxis.set_major_formatter(tick_formatter)
-
-        if len(values) > 1:
-            # two-way PDP - y-axis deciles + labels
-            deciles = mquantiles(X[:, fx[1]], prob=np.arange(0.1, 1.0, 0.1))
-            trans = transforms.blended_transform_factory(ax.transAxes,
-                                                         ax.transData)
-            xlim = ax.get_xlim()
-            ax.hlines(deciles, [0], 0.05, transform=trans, color='k')
-            ax.set_ylabel(name[1])
-            # hline erases xlim
-            ax.set_xlim(xlim)
-        else:
-            ax.set_ylabel('Partial dependence')
-
-        if len(values) == 1:
-            ax.set_ylim(pdp_lim[1])
-        axs.append(ax)
+    deciles = {}
+    for fx in chain.from_iterable(features):
+        if fx not in deciles:
+            deciles[fx] = mquantiles(X[:, fx], prob=np.arange(0.1, 1.0, 0.1))
 
-    fig.subplots_adjust(bottom=0.15, top=0.7, left=0.1, right=0.95, wspace=0.4,
-                        hspace=0.3)
+    display = PartialDependenceDisplay(pd_results, features, feature_names,
+                                       target_idx, pdp_lim, deciles)
+    return display.plot(ax=ax, n_cols=n_cols, line_kw=line_kw,
+                        contour_kw=contour_kw, fig=fig)
```

Comment:
Regarding `fig`, could we set `display.figure_` instead avoiding to have the `fig` in the new class. We can raise the deprecation warning in this function directly as well.

Reply:
Regarding the introduction of `fig`